src/WebJobs.Script.WebHost/Metrics/LinuxContainerMetricsPublisher.cs

```diff
@@ -0,0 +1,227 @@
+                _standbyOptionsOnChangeSubscription = _standbyOptions.OnChange(o => OnStandbyOptionsChange());
+            }
+            else
+            {
+                StartMetricsPublisher();
+            }
+        }
+
+        private void OnStandbyOptionsChange()
+        {
+            if (!_standbyOptions.CurrentValue.InStandbyMode)
+            {
+                StartMetricsPublisher();
+            }
+        }
+
+        public void AddFunctionExecutionActivity(string functionName, string invocationId, int concurrency, string executionStage, bool success, long executionTimeSpan, DateTime eventTimeStamp)
+        {
+            FunctionActivity activity = new FunctionActivity
+            {
+                FunctionName = functionName,
+                InvocationId = invocationId,
+                Concurrency = concurrency,
+                ExecutionStage = executionStage,
+                IsSucceeded = success,
+                ExecutionTimeSpanInMs = executionTimeSpan,
+                EventTimeStamp = eventTimeStamp,
+                Tenant = _environment.GetEnvironmentVariable(EnvironmentSettingNames.WebSiteStampDeploymentId)?.ToLowerInvariant()
+            };
+
+            if (!_functionActivities.TryAdd(activity))
+            {
+                _logTraceEvent(LogLevel.Warning, string.Format("LinuxContainerMetricsPublisher: Buffer full. Dropping current batch of function activities"));
+                DrainActivities(_currentFunctionActivities, _functionActivities);
+            }
+
+            _logTraceEvent(LogLevel.Information, string.Format("LinuxContainerMetricsPublisher: Added function activity : {0} {1} {2} {3} {4} {5}", functionName, invocationId, concurrency, executionStage, success, executionTimeSpan));
+        }
+
+        public void AddMemoryActivity(DateTime timeStampUtc, long data)
+        {
+            var memoryActivity = new MemoryActivity
+            {
+                CommitSizeInBytes = data,
+                EventTimeStamp = timeStampUtc,
+                Tenant = _environment.GetEnvironmentVariable(EnvironmentSettingNames.WebSiteStampDeploymentId)?.ToLowerInvariant()
+            };
+
+            if (!_memoryActivities.TryAdd(memoryActivity))
+            {
+                _logTraceEvent(LogLevel.Warning, string.Format("LinuxContainerMetricsPublisher: Buffer full. Dropping current batch of memory activities"));
+                DrainActivities(_currentMemoryActivities, _memoryActivities);
+            }
+        }
+
+        private void DrainActivities<T>(ConcurrentQueue<T> currentActivites, BlockingCollection<T> activitiesToProcess)
+        {
+            currentActivites.Clear();
+
+            if (activitiesToProcess.Any())
+            {
+                while (activitiesToProcess.TryTake(out T item))
+                {
+                    currentActivites.Enqueue(item);
+                }
+            }
+        }
+
+        internal void OnFunctionMetricsPublishTimer(object state)
```

Comment:
Make this "async void" so the publish activities can be properly async, without the GetAwaiter().GetResult()s